libs/threading_base/include/hpx/threading_base/scheduler_base.hpp

```diff
@@ -319,7 +342,8 @@ namespace hpx { namespace threads { namespace policies {
 
         std::atomic<std::int64_t> background_thread_count_;
 
-        util::function_nonser<void()> user_polling_function_;
+        polling_function_ptr polling_function_mpi_;
+        polling_function_ptr polling_function_cuda_;
```

Comment:
Despite what I said while we talked last, I think these should be atomics to avoid partial reads/writes in concurrent scenarios (on weak-memory platforms). However, you should be able to read and write using `std::memory_order_relaxed` as the pointers themselves are stable.